### PR TITLE
fix(subscriberdb): Detach inactive subscribers when syncing

### LINKED ISSUE #13461: [Regression Issue]: UE is not getting de-registered when we are making subscriber state to Inactive in NMS dashboard.
- Version: 1.8.0-1658998712-78900dda
- Affected Component: Access Gateway, or NMS
- Deployment Environment: e.g. Bare-metal (AGW) - VM based.

While performing Registration with PDU and Traffic when UE is active and PDU is established and from NMS if for registered IMSI we turn configs to Inactive, it is expected to trigger Deregistration after AGW receiving Inactive configuration from Orc8r. But still the IMSI is not getting deregistered.

ACTIVE -- IMSI001019976545660

INACTIVE -- IMSI001019976545660 

Please find the attached logs for the same:
[https://app.zenhub.com/files/170803235/b9265b46-8cc8-4d3c-a88f-091b2de2b309/download](https://app.zenhub.com/files/170803235/b9265b46-8cc8-4d3c-a88f-091b2de2b309/download)
[https://app.zenhub.com/files/170803235/cb063d5a-ed52-4533-b1a2-9392d0a44ba8/download](https://app.zenhub.com/files/170803235/cb063d5a-ed52-4533-b1a2-9392d0a44ba8/download)
[https://app.zenhub.com/files/170803235/a6e97268-8152-4d42-bfbb-ff4c746b0d37/download](https://app.zenhub.com/files/170803235/a6e97268-8152-4d42-bfbb-ff4c746b0d37/download)

### PR DESCRIPTION
**Pull request by sebathomas**
## Summary

Fixes #13461.

## Test Plan

* Set subscribers to INACTIVE in local NMS and observed deletion requests in local MME.
* Adapted an existing unit test to include an inactive subscriber.

## Additional Information

I originally tried to trigger the detach only if the state switched from ACTIVE to INACTIVE. However, putting this logic into the upsert function is a little ugly, and not putting it there would need additional DB queries. Additionally, we always trigger a detach on subscriber deletion and don't have any activity checks there either. So I think it's okay to skip this check and always do the detach. MME logs an error if the subscriber is not attached but it doesn't seem to cause problems.

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/python/magma/subscriberdb/client.py [hunk 1]
```diff
@@ -175,11 +176,17 @@ async def _sync_subscribers(self) -> bool:
         if not res.resync:
             # TODO(hcgatewood): switch to bulk editing subscriber data rows
             changeset = self._process_changeset(res.changeset)
+            inactive = []
             for subscriber_data in changeset.to_renew:
                 self._store.upsert_subscriber(subscriber_data)
+                if (
```
Comment (crasu) [inline lte/gateway/python/magma/subscriberdb/client.py]: May be this should go to a seperate function with its own loop/list comprehension.

File: lte/gateway/python/magma/subscriberdb/client.py [hunk 2]
```diff
@@ -179,7 +180,13 @@ async def _sync_subscribers(self) -> bool:
...
             for sid in changeset.deleted:
                 self._store.delete_subscriber(sid)
-            self._detach_subscribers_by_ids(changeset.deleted)
+
+            inactive = [
+                SIDUtils.to_str(subscriber.sid)
+                for subscriber in changeset.to_renew
+                if (subscriber.lte and subscriber.lte.state != LTESubscription.ACTIVE)
+            ]
+            self._detach_subscribers_by_ids(list(changeset.deleted) + inactive)
```
Comment (MoritzThomasHuebner) [inline lte/gateway/python/magma/subscriberdb/client.py:189]: `changeset.deleted` should already be a list, so we don't need to wrap it in another `list(...)`?

Comment (sebathomas) [reply lte/gateway/python/magma/subscriberdb/client.py:189]: The typing of the protobuf objects is a little weird. `changeset.deleted` is actually a RepeatedScalarContainer and you get a runtime error if you try to add elements without converting into a list.

Comment (MoritzThomasHuebner) [reply lte/gateway/python/magma/subscriberdb/client.py:189]: Interesting!

**Review by voisey — APPROVED** [review]

**Pull request merged**